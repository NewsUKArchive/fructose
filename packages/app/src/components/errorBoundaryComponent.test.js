/* globals it expect */

import React from "react";
import { Text } from "react-native";
import { EventEmitter } from "events";
import renderer from "react-test-renderer";

import ErrorBoundary from "./errorBoundaryComponent";

const ThrowAnError = () => <Text>{lol}</Text>; // eslint-disable-line

describe("Error Boundary", () => {
  it("It renders the error View when error is triggered", () => {
    const wrapper = renderer
      .create(
        <ErrorBoundary socket={new EventEmitter()} events={new EventEmitter()}>
          <ThrowAnError />
        </ErrorBoundary>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("it render a component when there is no error", () => {
    const wrapper = renderer
      .create(
        <ErrorBoundary socket={new EventEmitter()} events={new EventEmitter()}>
          <Text>Hey I rendered</Text>
        </ErrorBoundary>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
