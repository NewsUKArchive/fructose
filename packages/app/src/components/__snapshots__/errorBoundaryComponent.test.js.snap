// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary It renders the error View when error is triggered 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#B20000",
      "height": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Exception Found
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    lol is not defined
  </Text>
</View>
`;

exports[`Error Boundary it render a component when there is no error 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Hey I rendered
</Text>
`;
